### PR TITLE
Add documentation for the messaging endpoint

### LINKED ISSUE #2063: Add documentation for the messaging/notification endpoint
## Suggested addition

Currently there is no documentation on how to interact with the messaging endpoint. It would be nice to add documentation for the endpoint that would help developers/users know how to interact with it.

### PR DESCRIPTION
**Pull request by DavisRayM**
### Changes / Features implemented

- Add documentation for the messaging viewset

### Steps taken to verify this change does what is intended

N/A

### Side effects of implementing this change

N/A

### Note for reviewers

I've tagged the documentation as WIP since the `project` and `user` target types are currently not utilized on the application; There is currently no message sent for those target types

Closes #2063

### CODE REVIEW

File: docs/messaging.rst [hunk 1]
```diff
@@ -0,0 +1,101 @@
+Messaging (WIP)
+***************
```
Comment (ukanga) [inline docs/messaging.rst:2]: What is still outstanding? Is there a plan to address that later?

Comment (DavisRayM) [reply docs/messaging.rst:2]: Documentation on how to utilizes the `project` and `user` target types is the only thing that was pending at the time; But it doesn't seem like we currently utilize those target types.

Comment (DavisRayM) [reply docs/messaging.rst:2]: The initial plan was to mention it for now and improve the documentation once the user and project target types are in use; _I'd like to think that they'd be used for the activity log functionality_

**Review by denniswambua — APPROVED** [review]

File: docs/messaging.rst [hunk 2]
```diff
@@ -0,0 +1,101 @@
+Messaging
+*********
+
+This endpoint provides access to event messages sent for a specific target. Where:
+
+- ``target_type`` - The type of the target object. The supported target types are ``xform``, ``project`` & ``user``.
+- ``target_id`` - The unique identifier of the target object.
+- ``verb`` - A specific action that has occured on the object i.e ``form_updated``
```
[the remark this answers is not in the document]
Comment (DavisRayM) [reply docs/messaging.rst:8]: Listed them here

https://github.com/onaio/onadata/pull/2026/files#diff-519863318bd120f48185680a0cc452213e95ae2b5c15793abc2513e4fc11daa7R67

I'll create a PR to list them out at the beginning of the document

**Pull request merged**